lib/src/provider/asset_picker_provider.dart

```diff
@@ -254,6 +254,155 @@ abstract class AssetPickerProvider<Asset, Path> extends ChangeNotifier {
     set.remove(item);
     selectedAssets = set;
   }
+
+  /// 拖拽状态
+  /// Drag status
+  bool isInDragging = false;
+
+  /// 边缘自动滚动控制器
+  /// Edge Auto Scrolling Detector. Use to support edge auto scroll when drag position reach the edge of device's screen.
+  EdgeDraggingAutoScroller? _autoScroller;
+
+  // An eyeballed value for a smooth scrolling experience.
+  static const double _kDefaultAutoScrollVelocityScalar = 50;
+
+  /// 起始选择序号
+  /// Item index of the first selected item
+  int initialSelectedIdx = -1;
+
+  /// 拖拽选择 或 拖拽取消选择
+  /// Drag to select or deselect state
+  bool dragSelect = true;
+
+  /// 长按启动拖拽
+  /// Long Press to enable drag and select
+  void onDragStart(
+    BuildContext context,
+    LongPressStartDetails details,
+    int index,
+    Asset entity,
+  ) {
+    isInDragging = true;
+
+    final scrollableState = _checkScrollableStatePresent(context);
+    if (scrollableState == null) {
+      return;
+    }
+
+    _autoScroller = EdgeDraggingAutoScroller(
+      scrollableState,
+      velocityScalar: _kDefaultAutoScrollVelocityScalar,
+    );
+
+    initialSelectedIdx = index;
+
+    dragSelect = !selectedAssets.contains(entity);
+  }
+
+  void onDragUpdate(
+    BuildContext context,
+    LongPressMoveUpdateDetails details,
+    double itemSize,
+    int gridCount,
+    double topPadding,
+  ) {
+    if (!isInDragging) {
+      return;
+    }
+
+    if (dragSelect && selectedAssets.length == maxAssets) {
+      return;
+    }
+
+    final scrollableState = _checkScrollableStatePresent(context);
+    if (scrollableState == null) {
+      return;
+    }
+
+    final column = _getIndex(details.globalPosition.dx, itemSize);
+    final row = _getIndex(
+      details.globalPosition.dy -
```

Comment:
Consider using `localPosition`.

Reply:
I will try to calculate the index through localPosition